vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleMainComponent.java

```diff
@@ -0,0 +1,140 @@
+      @Autowired ExperimentInformationService experimentInformationService,
+      @Autowired SampleOverviewComponent sampleOverviewComponent) {
+    this.addClassName("main");
+    this.sampleOverviewComponent = sampleOverviewComponent;
+    this.projectInformationService = projectInformationService;
+    this.experimentInformationService = experimentInformationService;
+    reloadOnBatchRegistration();
+  }
+
+  /**
+   * Provides the {@link ProjectId} to the components within this container
+   * <p>
+   * This method serves as an entry point providing the necessary {@link ProjectId} to components
+   * within this container
+   *
+   * @param projectId projectId of the selected project
+   */
+  public void projectId(ProjectId projectId) {
+    this.projectId = projectId;
+    projectInformationService.find(projectId)
+        .ifPresentOrElse(project -> {
+          propagateProjectInformation(projectId);
+          propagateExperimentInformation(projectId);
+        }, this::displayProjectNotFound);
+  }
+
+  private void propagateProjectInformation(ProjectId projectId) {
+    sampleOverviewComponent.setProject(projectId);
+  }
+
+  private void propagateExperimentInformation(ProjectId projectId) {
+    Collection<Experiment> experiments = getExperimentsForProject(projectId);
+    if (experiments.isEmpty()) {
+      displayNoExperimentsFound();
+    } else {
+      displayComponentInContent(sampleOverviewComponent);
+      sampleOverviewComponent.setExperiments(experiments);
+    }
+  }
+
+  private Collection<Experiment> getExperimentsForProject(ProjectId projectId) {
+    var project = projectInformationService.find(projectId);
+    return project.<Collection<Experiment>>map(value -> value.experiments().stream()
+        .map(experimentInformationService::find).filter(Optional::isPresent).map(Optional::get)
+        .toList()).orElseGet(ArrayList::new);
+  }
+
+  private boolean isComponentInContent(Component component) {
+    return this.getChildren().collect(Collectors.toSet()).contains(component);
+  }
+
+  private void displayComponentInContent(Component component) {
+    if (!isComponentInContent(component)) {
+      this.removeAll();
+      this.add(component);
+    }
+  }
+
+  /**
+   * Propagates the listener which will retrieve notification if a new {@link Batch} was created in
+   * the {@link SampleOverviewComponent} within this container
+   *
+   * @param batchRegistrationListener listener to be notified if a batchRegistration happened in the
+   *                                  {@link SampleOverviewComponent}
+   */
+  public void addBatchRegistrationListener(BatchRegistrationListener batchRegistrationListener) {
+    sampleOverviewComponent.addBatchRegistrationListener(batchRegistrationListener);
+  }
+
+  private void reloadOnBatchRegistration() {
+    sampleOverviewComponent.addBatchRegistrationListener(
+        event -> propagateExperimentInformation(projectId));
+  }
```

Comment:
This is a bit confusing: you are setting up a listener in the method that suggests reloading takes place. let's maybe discuss this on monday